tests/data/drivers/test_sumologic_driver.py

```diff
@@ -0,0 +1,219 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""datq query test class."""
+import io
+
+from unittest.mock import patch, MagicMock
+import pytest
+import pytest_check as check
+
+import pandas as pd
+
+from msticpy.common.exceptions import (
+    MsticpyUserConfigError,
+    MsticpyConnectionError,
+    MsticpyNotConnectedError,
+)
+
+from msticpy.data.drivers.sumologic_driver import SumologicDriver, sp_client
+
+from ...unit_test_lib import get_test_data_path
+
+_TEST_DATA = get_test_data_path()
+
+SUMOLOGIC_CLI_PATCH = SumologicDriver.__module__ + ".sp_client"
+SUMOLOGIC_RESULTS_PATCH = SumologicDriver.__module__ + ".sp_results"
+
+
+# pylint: disable=too-many-branches, too-many-return-statements
+# pylint: disable=no-self-use, too-few-public-methods, protected-access
+
+
+def cli_connect(**kwargs):
+    """Return None if magic isn't == kql."""
+    cause = MagicMock()
+    cause.body = bytes("Test body stuff", encoding="utf-8")
+    cause.status = 404
+    cause.reason = "Page not found."
+    cause.headers = "One Two Three"
+    if kwargs.get("host") == "AuthError":
+        raise sp_client.AuthenticationError(cause=cause, message="test AuthHeader")
+    if kwargs.get("host") == "HTTPError":
+        cause.body = io.BytesIO(cause.body)
+        raise sp_client.HTTPError(response=cause, _message="test HTTPError")
+    return _MockSumologicService()
+
+
+class _MockSumologicSearch:
+    def __init__(self, name, search):
+        self.name = name
+        self.search = search
+
+    def get(self, arg):
+        """Mock method."""
+        del arg
+        return self.search
+
+    def __getitem__(self, key):
+        """Mock method."""
+        if key == "search":
+            return self.search
+        return "other"
+
+
+class _MockAlert:
+    def __init__(self, name, count):
+        self.name = name
+        self.count = count
+
+
+class _MockSumologicService(MagicMock):
```

Comment:
See the updates I did (partially) in the new branch.
The mock has to replicate the SumoLogic class otherwise it isn't doing it's job.
You need to implement search_job, search_job_status and search_job_records.
- search_job - this should take the params and return an emulated job object
- search_job_status - returns different results (you can drive the expected result status by setting _MockSumologicService.status to the desired value.
- search_job_records - should return json objects emulating SL results and check that they are correctly converted into the expect DF.